index.js

```diff
@@ -45,6 +45,30 @@ module.exports = {
   rules: {
     // open a PR per rule change
 
+    // Simplifies code and reduces bugs
+    // https://eslint.org/docs/rules/no-lonely-if
+    'no-lonely-if': ['error'],
+
+    // Reduces careless typo mistakes
+    // https://eslint.org/docs/rules/no-return-assign
+    'no-return-assign': ['error'],
+
+    // Spreading is better than complex Object.assign
+    // https://eslint.org/docs/rules/prefer-object-spread
+    'prefer-object-spread': ['error'],
+
+    // Not all arrow functions need to return
+    // https://eslint.org/docs/rules/consistent-return
+    'consistent-return': 'off',
+
+    // This catches typos that end up being bitwise operator
+    // However, some operators are common and should be allowed
+    // https://eslint.org/docs/rules/no-bitwise
+    'no-bitwise': ['error', { allow: ['~'] }],
+
+    // AirBnB has this as an error but that's disruptive during development
+    'no-unused-vars': 'warn',
```

Comment:
It might be disruptive during development but, there are many who would ignore warns too much in this case which would be detrimental.

Reply:
agree with this point

Reply:
There's a better solution for that:

```javascript
  "lint-staged": {
    "src/**/*.{js,json}": [
      "eslint --fix --max-warnings=0",
    }
}
```

Reply:
That one change to package.json prevents warnings from being committed, since that's what you're worried about.

Reply:
However, I'm going to give you a reason why this should be warn and not error and not to add that max warnings.

We live in a world where computers break and things go wrong. It is insanity to block people from pushing work in progress because they have unused vars.

If you want to figure out a way to make it so no work can be merged into master if it has unused vars, that sounds great. But on branches, all you're doing is taunting the programming gods and expecting not to get punished. It's foolish.

If you decide to tempt the fates and leave this rule as "error", I and every other rational developer who works with this rule set will override it to warn, or just turn it off. Just saying. Don't think Murphy isn't out to get you. He is.

Reply:
If git hooks are used to run eslint, work in progress can be committed via `--no-verify`. That's an opt-in to doing the dangerous thing (still a warning for a reason) and keeps the pit of success.

Reply:
Sometimes, you need to be strict (error). Sometimes, a gentle reminder is all you need (warn).

This rule reduces cruft, and that's a good goal. It doesn't need to be an error, though. A warning is sufficient.

If you have developers who consistently write code where they are ignoring this warning, then that's a different issue that should be resolved outside of a ruleset.